base_local_planner/src/speed_limiters/static_object_speed_limiter.cpp

```diff
@@ -0,0 +1,423 @@
+  {
+    SpeedLimiterData data;
+    data.dist_left_ = dist_left;
+    data.dist_right_ = dist_right;
+    data.speed_ = speed;
+    data.min_velocity_ = params_.min_linear_velocity;
+    data.min_test_distance_ = params_.min_linear_velocity_distance;
+    data.max_test_distance_ = params_.max_linear_velocity_distance;
+    data.min_test_velocity_ = params_.min_linear_velocity_test_speed;
+    data.max_test_velocity_ = params_.max_linear_velocity_test_speed;
+    data.min_reduction_ = params_.min_linear_velocity_reduction;
+    data.max_reduction_ = params_.max_linear_velocity_reduction;
+
+    return calculateAllowedSpeed(data);
+  }
+
+  StaticObjectSpeedLimiter::SpeedLimiterResult StaticObjectSpeedLimiter::calculateAllowedAngularSpeed(
+      const double dist_left, const double dist_right, const double speed) const
+  {
+    SpeedLimiterData data;
+    data.dist_left_ = dist_left;
+    data.dist_right_ = dist_right;
+    data.speed_ = speed;
+    data.min_velocity_ = params_.min_angular_velocity;
+    data.min_test_distance_ = params_.min_angular_velocity_distance;
+    data.max_test_distance_ = params_.max_angular_velocity_distance;
+    data.min_test_velocity_ = params_.min_angular_velocity_test_speed;
+    data.max_test_velocity_ = params_.max_angular_velocity_test_speed;
+    data.min_reduction_ = params_.min_angular_velocity_reduction;
+    data.max_reduction_ = params_.max_angular_velocity_reduction;
+
+    return calculateAllowedSpeed(data);
+  }
+
+  StaticObjectSpeedLimiter::SpeedLimiterResult
+  StaticObjectSpeedLimiter::calculateAllowedSpeed(const SpeedLimiterData &data) const
+  {
+    SpeedLimiterResult result;
+
+    if (data.speed_ < data.min_test_velocity_)
+    {
+      return result;
+    }
+
+    double result_speed = data.speed_;
+
+    double size_delta = (data.speed_ - data.min_test_velocity_) / (data.max_test_velocity_ - data.min_test_velocity_);
+    size_delta *= (data.max_test_distance_ - data.min_test_distance_);
+
+    double test_size = data.max_test_distance_;
+    if (params_.test_distance_changes_with_speed)
+    {
+      if (params_.test_distance_grows_with_speed)
+      {
+        test_size = size_delta + data.min_test_distance_;
+      }
+      else
+      {
+        test_size = data.max_test_distance_ - size_delta;
+      }
+      test_size = std::max(std::min(data.max_test_distance_, test_size), data.min_test_distance_);
+    }
+
+    if (data.dist_left_ >= data.min_test_distance_ && data.dist_left_ <= data.max_test_distance_)
+    {
+      if (data.dist_left_ <= test_size)
+      {
+        double speed_delta = twoLevelInterpolation(test_size, data.min_test_distance_, data.max_test_distance_, data.min_reduction_,
+                                                   data.max_reduction_);
+        if ((data.speed_ - speed_delta >= data.min_velocity_) && (result_speed > data.speed_ - speed_delta))
+        {
+          result_speed = data.speed_ - speed_delta;
+        }
```

Comment:
seems like this chunk is a clone of the "dist right" chunk right below it? could it be pulled out into its own method?